FindmallocMC.cmake

```diff
@@ -97,44 +102,47 @@ if(mallocMC_ROOT_DIR)
     # mallocMC variables ########################################################
     #
     set(mallocMC_VERSION "${mallocMC_VERSION_MAJOR}.${mallocMC_VERSION_MINOR}.${mallocMC_VERSION_PATCH}")
-    set(mallocMC_INCLUDE_DIRS ${mallocMC_ROOT_DIR}/include)
+    set(mallocMC_INCLUDE_DIRS ${mallocMC_ROOT_DIR}/include CACHE PATH "mallocMC include pathes")
 
     # check additional components ###############################################
     #
     foreach(COMPONENT ${mallocMC_FIND_COMPONENTS})
-        set(mallocMC_${COMPONENT}_FOUND TRUE)
+        set(mallocMC_${COMPONENT}_FOUND TRUE CACHE BOOL
+            "The mallocMC component ${COMPONENT} exists in the found installation.")
 
         if(${COMPONENT} STREQUAL "halloc")
 
             # halloc linked library #################################################
             #
             list(APPEND mallocMC_REQUIRED_VARS_LIST mallocMC_LIBRARIES)
-            find_library(${COMPONENT}_LIBRARY
+            find_library(mallocMC_${COMPONENT}_LIBRARY
                 NAMES libhalloc.a
                 PATHS "${mallocMC_ROOT_DIR}/../halloc/" ENV HALLOC_ROOT
                 PATH_SUFFIXES "lib" "bin"
-                )
-            if(${COMPONENT}_LIBRARY)
-                list(APPEND mallocMC_LIBRARIES ${${COMPONENT}_LIBRARY})
-            else(${COMPONENT}_LIBRARY)
+                DOC "Libraries for the mallocMC component ${COMPONENT}."
+            )
+            if(mallocMC_${COMPONENT}_LIBRARY)
+                list(APPEND mallocMC_LIBRARIES ${mallocMC_${COMPONENT}_LIBRARY})
+            else(mallocMC_${COMPONENT}_LIBRARY)
                 if(mallocMC_FIND_REQUIRED OR NOT mallocMC_FIND_QUIETLY)
                     message(WARNING "libhalloc.a not found. Ensure it is compiled correctly and set HALLOC_ROOT")
                 endif()
-                set(mallocMC_${COMPONENT}_FOUND FALSE)
-            endif(${COMPONENT}_LIBRARY)
+                unset(mallocMC_${COMPONENT}_FOUND CACHE)
+            endif(mallocMC_${COMPONENT}_LIBRARY)
 
             # halloc headers ########################################################
             #
-            find_path(${COMPONENT}_INCLUDE_DIR
+            find_path(mallocMC_${COMPONENT}_INCLUDE_DIR
                 NAMES halloc.h
                 PATHS "${mallocMC_ROOT_DIR}/../halloc/" ENV HALLOC_ROOT
                 PATH_SUFFIXES "include" "src"
-                )
-            if(${COMPONENT}_INCLUDE_DIR)
-                list(APPEND mallocMC_INCLUDE_DIRS ${${COMPONENT}_INCLUDE_DIR})
-            else(${COMPONENT}_INCLUDE_DIR)
-                set(mallocMC_${COMPONENT}_FOUND FALSE)
-            endif(${COMPONENT}_INCLUDE_DIR)
+                DOC "Includes for the mallocMC component ${COMPONENT}."
+            )
+            if(mallocMC_${COMPONENT}_INCLUDE_DIR)
+                list(APPEND mallocMC_INCLUDE_DIRS ${mallocMC_${COMPONENT}_INCLUDE_DIR})
```

Comment:
It looks like this call to `list(APPEND...)` does **not** do anything, presumably because it can not append to a cache variable?
`mallocMC_halloc_INCLUDE_DIR = /home/carli/dev/picongpu/thirdParty/mallocMC/halloc/src`
but in the end,
`mallocMC_INCLUDE_DIRS = /home/carli/dev/picongpu/thirdParty/mallocMC/src/include` (the new include_dir is missing).
Probably this has to be done using FORCE to update the cached variable?